Source/WTF/wtf/text/StringImpl.cpp

```diff
@@ -325,17 +325,14 @@ Ref<StringImpl> StringImpl::create8BitIfPossible(std::span<const char16_t> chara
     if (characters.empty())
         return *empty();
 
-    std::span<Latin1Character> data;
-    auto string = createUninitializedInternalNonEmpty(characters.size(), data);
-
-    size_t i = 0;
-    for (auto character : characters) {
-        if (!isLatin1(character))
-            return create(characters);
-        data[i++] = static_cast<Latin1Character>(character);
+    if (std::ranges::all_of(characters, [](auto c) { return isLatin1(c); })) {
```

Comment:
This sounds like something that would benefit from SIMD? or maybe using `charactersAreAllLatin1`
```suggestion
    if (charactersAreAllLatin1(characters)) {
```

Reply:
`charactersAreAllLatin1()` sounds good. It is simpler. I'll re-do perf A/B testing with this suggestion to make sure the win is still there.